### DISCUSSION TITLE
Change BLE type Server - Client

### DISCUSSION THREAD

**Discussion by mauroForlimpopoli** [Moddable-OpenSource/moddable#696] [Q&A]
In my project I start the system with BLE uart-server. Can I change to BLE uart-client without restar the system ? Regards

**Comment by phoddie** [comment]
Maybe it isn't necessary to try switching. The only difference between the UART client and server is which one initiates the connection. Both ends of the connection can send and receive data.

**Comment by mauroForlimpopoli** [comment]
Like your told me before, when I use the bluetooth serial es:
(let server = new UARTServer;)
the system uses about 30kB of RAM.
When the transmission is ended, can I disconnect the device and to release the 30kB of RAM?
regards

**Comment by phoddie** [comment]
Both the BLE client and server instances provide a `close` method. You can call that. The amount of memory that is freed depends on the details of the host implementation. I cannot say how much memory may be recovered in your situation.

**Reply by mauroForlimpopoli** [reply]
I tried with :
server.close()
it releases about 30 kB
Thank you

**Reply by phoddie** [reply]
Good news.

**Reply by mauroForlimpopoli** [reply]
Good morning,
when the device is "Server" and the connection is "On" how can I disconnect the service device ?
I tryied with  server.close(), but the connection stay "on".
Regards
